### PR TITLE
4x alpha release process

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/ChainSafe/web3.js/issues/5067

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by Muhammad-Altabba — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -20,25 +20,30 @@ Further details about versioning can be found in the [semver 2.0.0 specification
...
     - In the release description, copy all entries in `CHANGELOG.md` for the version being released
-12. Click `Save draft`
-13. Open pull request to merge branch created in `Step 2` (`release/package-name/bumped-version`) into `4.x`
-14. When sufficient approvals have been met, merge the pull request
+10. Click `Save draft`
+11. Open pull request to merge branch created in `Step 2` (`release/bumped-version`) into `4.x`
+12. Wait for all tests to pass in github CI/CD actions
+13. When sufficient approvals have been met, merge the pull request
+14. Publish documentation changes
 15. Publish draft release created in `Step 11`
```
Comment (Spacesai1or) [inline RELEASE.md]:
```suggestion
15. Publish draft release created in `Step 9`
```

File: RELEASE.md [hunk 2]
```diff
@@ -20,25 +20,30 @@ Further details about versioning can be found in the [semver 2.0.0 specification
...
-16. Run `npm publish` in the package directory to publish to NPM
-    - If a tag is needed, such as `alpha`, use `npm publish --tag alpha`
+16. For each package, go in package and run `yarn publish --tag alpha` to publish to NPM
```
Comment (Spacesai1or) [inline RELEASE.md]: Couldn't this be done using something like `lerna publish from-package --dist-tag alpha`?

**Review by avkos — APPROVED** [review]

File: RELEASE.md (hunk 2 above)
Comment (jdevcs) [reply RELEASE.md]: I was thinking to :
- minimum use lerna as we will remove it in future, 
- and lerna publish invokes life cycle scripts ( including version, tags push etc ) but we can avoid that using  --ignore-scripts

But its good we can use it for now as by using lerna here we dnt need to take care packages topological order during publishing.

File: RELEASE.md [hunk 3]
```diff
@@ -20,25 +20,30 @@ Further details about versioning can be found in the [semver 2.0.0 specification
 
 ## Release Process
 
-### Manual Steps (needs to be done per package)
+### Manual Steps For Alpha Release
 
 1. `git checkout 4.x`: Verify you are on the `4.x` base branch
-2. `git checkout -b release/package-name/bumped-version`: Checkout a branch with the `package-name` and `bumped-version` e.g. `git checkout -b release/web3-packagetemplate/1.0.1`
+2. `git checkout -b release/bumped-version`: Create and checkout a branch with the `bumped-version` e.g. `git checkout -b release/4.0.0-alpha.0`
+    - `bumped-version` of release branch should be of main web3 package.
```
Comment (Spacesai1or) [inline RELEASE.md:27]: I'm not sure what you mean by this

**Review by Spacesai1or — APPROVED** [review]

File: RELEASE.md (hunk 3 above)
Comment (jdevcs) [reply RELEASE.md:27]: - `bumped-version` of release branch should be of main web3 package. 

is for release branch naming as we will have independent versioning so initially we can name release branch same like our main web3 package.

**Pull request merged**